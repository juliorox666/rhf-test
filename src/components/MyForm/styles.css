.myForm {
  max-width: 1200px;
  padding: 40px;
}

.myForm__title {
  font-size: 24px;
  font-weight: bold;
  color: #000;
  padding: 18px;
  width: 100%;
  box-sizing: border-box;
  display: flex;
  justify-content: center;
}

.myForm__box {
  padding: 40px;
  border: 1px solid #dedede;
  border-radius: 4px;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
}

.formData {
  width: 100%;
}

.formData__label {
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  padding: 16px 0;
  align-items: center;
}
.formData__text {
  font-size: 18px;
  padding: 0 18px;
}

.formData__submit {
  display: flex;
  justify-content: center;
}
.submit__button {
  font-size: 18px;
  width: 200px;
  color: #fff;
  margin: 16px 0;
  padding: 16px;
  background-color: #1976d2;
  box-shadow: 0px 3px 1px -2px rgb(0 0 0 / 20%),
    0px 2px 2px 0px rgb(0 0 0 / 14%), 0px 1px 5px 0px rgb(0 0 0 / 12%);
  border: none;
}
