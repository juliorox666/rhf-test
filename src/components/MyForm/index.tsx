/* eslint-disable react/prop-types */
import React, { useEffect, useState } from "react";
import { styled } from "@material-ui/core/styles";
import TextField, { TextFieldProps } from "@material-ui/core/TextField";
import Alert from "@material-ui/lab/Alert";
import { useForm } from "react-hook-form";
import "./styles.css";

const TextFieldCustom = styled(TextField)({
  border: 0,
  borderRadius: 3,
  color: "white",
});

type InputTextProps = {
  id: string;
  "data-testid": string;
  name: string;
  error: boolean;
  helperText: React.ReactNode;
  label: string;
};
const InputText = React.forwardRef<TextFieldProps, InputTextProps>(
  ({ "data-testid": datatestid, ...props }: InputTextProps, ref) => (
    <TextFieldCustom
      // eslint-disable-next-line react/jsx-props-no-spreading
      {...props}
      inputRef={ref}
      inputProps={{ "data-testid": datatestid }}
      variant="outlined"
    />
  )
);

interface User {
  firstName: string;
  lastName: string;
  phone: string;
  email: string;
  gender: string;
}

interface MyFormProps {
  onSubmitHandler: (data: User) => void;
}

const MyForm: React.FC<MyFormProps> = ({
  onSubmitHandler = (data) => JSON.stringify(data),
}: MyFormProps) => {
  const [user, setUser] = useState<User | null>();
  const {
    register,
    handleSubmit,
    errors,
    formState: { isDirty },
    reset,
  } = useForm<User>();
  const onSubmit = handleSubmit((data) => {
    onSubmitHandler(data);
    setUser(data);
  });

  useEffect(() => {
    if (user) {
      reset(user);
    }
  }, [reset, user]);

  return (
    <div className="my-form">
      <pre>
        <code>{JSON.stringify(user, null, 4)}</code>
      </pre>
      <div className="my-form__message">
        {user?.firstName && (
          <Alert severity="success">Sign up successfully done!</Alert>
        )}
      </div>
      <div className="my-form__title">üìù Register Form</div>
      <div className="my-form__box">
        <form className="form-data" onSubmit={onSubmit}>
          <div className="form-data__label">
            <InputText
              error={!!errors.firstName}
              helperText={
                errors.firstName ? (
                  <span data-testid="firstname-error-message">
                    First name is required
                  </span>
                ) : (
                  ``
                )
              }
              label="First name"
              data-testid="firstName"
              id="firstName"
              name="firstName"
              ref={register({ required: true })}
            />
          </div>
          <div className="form-data__label">
            <InputText
              error={!!errors.lastName}
              helperText={
                errors.lastName ? (
                  <span data-testid="lastname-error-message">
                    Last name is required
                  </span>
                ) : (
                  ``
                )
              }
              label="Last name"
              data-testid="lastName"
              id="lastName"
              name="lastName"
              ref={register({ required: true })}
            />
          </div>
          <div className="form-data__label">
            <InputText
              error={!!errors.phone}
              helperText={errors.phone ? `Phone is required` : ``}
              label="Phone"
              data-testid="phone"
              id="phone"
              name="phone"
              ref={register({ required: true })}
            />
          </div>
          <div className="form-data__label">
            <InputText
              error={!!errors.email}
              helperText={errors.email ? `Email is required` : ``}
              label="Email"
              data-testid="email"
              id="email"
              name="email"
              ref={register({ required: true })}
            />
          </div>
          <div className="form-data__label">
            <select
              className="form-data__label select-component"
              name="gender"
              data-testid="gender"
              ref={register({ required: true })}
            >
              <option value="">- SELECT ONE -</option>
              <option value="male">Male</option>
              <option value="female">Female</option>
              <option value="other">Other</option>
            </select>
            {errors.gender && <span role="alert">Gender is required</span>}
          </div>
          <div className="form-data__submit submit">
            <input
              disabled={!isDirty}
              id="submit"
              data-testid="submit"
              className="submit__button"
              type="submit"
              value="Submit"
            />
          </div>
        </form>
      </div>
    </div>
  );
};

export default MyForm;
