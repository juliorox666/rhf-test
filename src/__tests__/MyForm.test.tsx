import React from "react";
import { render, act, fireEvent } from "@testing-library/react";
import MyForm from "components/MyForm";

describe("Form validation", () => {
  test("should inputs change values", () => {
    const mockOnSubmit = jest.fn();
    const { getByTestId } = render(<MyForm onSubmitHandler={mockOnSubmit} />);

    const firstName = getByTestId("firstName") as HTMLInputElement;
    const lastName = getByTestId("lastName") as HTMLInputElement;

    act(() => {
      fireEvent.change(firstName, { target: { value: "Julio" } });
      fireEvent.change(lastName, { target: { value: "Guedes" } });
    });

    expect(firstName).toHaveValue("Julio");
    expect(lastName).toHaveValue("Guedes");
  });

  test("should firstName be required for submitting", async () => {
    const mockOnSubmit = jest.fn();
    const { getByTestId, container } = render(
      <MyForm onSubmitHandler={mockOnSubmit} />
    );

    const firstName = getByTestId("firstName") as HTMLInputElement;
    const lastName = getByTestId("lastName") as HTMLInputElement;
    const submit = getByTestId("submit");

    await act(async () => {
      fireEvent.input(firstName, { target: { value: "" } });
    });
    await act(async () => {
      fireEvent.input(lastName, { target: { value: "Guedes" } });
    });

    await act(async () => {
      fireEvent.click(submit);
    });

    expect(container).toMatchSnapshot();
  });

  test("should lastName be required for submitting", async () => {
    const mockOnSubmit = jest.fn();
    const { getByTestId, container } = render(
      <MyForm onSubmitHandler={mockOnSubmit} />
    );

    const firstName = getByTestId("firstName") as HTMLInputElement;
    const lastName = getByTestId("lastName") as HTMLInputElement;
    const submit = getByTestId("submit");

    await act(async () => {
      fireEvent.input(firstName, { target: { value: "Julio" } });
    });
    await act(async () => {
      fireEvent.input(lastName, { target: { value: "" } });
    });

    await act(async () => {
      fireEvent.click(submit);
    });

    expect(container).toMatchSnapshot();
  });

  test("should be able to submit the form", async () => {
    const mockOnSubmit = jest.fn();
    const { getByTestId, container } = render(
      <MyForm onSubmitHandler={mockOnSubmit} />
    );

    const firstName = getByTestId("firstName") as HTMLInputElement;
    const lastName = getByTestId("lastName") as HTMLInputElement;
    const phone = getByTestId("phone") as HTMLInputElement;
    const email = getByTestId("email") as HTMLInputElement;
    const gender = getByTestId("gender") as HTMLSelectElement;

    const submit = getByTestId("submit");

    await act(async () => {
      fireEvent.change(firstName, { target: { value: "Julio" } });
      fireEvent.change(lastName, { target: { value: "Guedes" } });
      fireEvent.change(phone, { target: { value: "31998282827" } });
      fireEvent.change(email, { target: { value: "julio@gmail.com" } });
      fireEvent.change(gender, { target: { value: "male" } });
    });

    await act(async () => {
      fireEvent.click(submit);
    });

    expect(mockOnSubmit).toHaveBeenCalledTimes(1);
    expect(container).toMatchSnapshot();
  });
});
