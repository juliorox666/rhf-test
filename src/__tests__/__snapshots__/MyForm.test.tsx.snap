// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form validation should be able to submit the form 1`] = `
<div>
  <div
    class="my-form"
  >
    <pre>
      <code>
        {
    "firstName": "Julio",
    "lastName": "Guedes",
    "phone": "31998282827",
    "email": "julio@gmail.com",
    "gender": "male"
}
      </code>
    </pre>
    <div
      class="my-form__message"
    >
      <div
        class="MuiPaper-root MuiAlert-root MuiAlert-standardSuccess MuiPaper-elevation0"
        role="alert"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        >
          Sign up successfully done!
        </div>
      </div>
    </div>
    <div
      class="my-form__title"
    >
      üìù Register Form
    </div>
    <div
      class="my-form__box"
    >
      <form
        class="form-data"
      >
        <div
          class="form-data__label"
        >
          <div
            class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-16"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink="true"
              for="firstName"
              id="firstName-label"
            >
              First name
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="firstName"
                id="firstName"
                name="firstName"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-19 PrivateNotchedOutline-legendNotched-20"
                >
                  <span>
                    First name
                  </span>
                </legend>
              </fieldset>
            </div>
            
          </div>
        </div>
        <div
          class="form-data__label"
        >
          <div
            class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-16"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink="true"
              for="lastName"
              id="lastName-label"
            >
              Last name
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="lastName"
                id="lastName"
                name="lastName"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-19 PrivateNotchedOutline-legendNotched-20"
                >
                  <span>
                    Last name
                  </span>
                </legend>
              </fieldset>
            </div>
            
          </div>
        </div>
        <div
          class="form-data__label"
        >
          <div
            class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-16"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink="true"
              for="phone"
              id="phone-label"
            >
              Phone
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="phone"
                id="phone"
                name="phone"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-19 PrivateNotchedOutline-legendNotched-20"
                >
                  <span>
                    Phone
                  </span>
                </legend>
              </fieldset>
            </div>
            
          </div>
        </div>
        <div
          class="form-data__label"
        >
          <div
            class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-16"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink="true"
              for="email"
              id="email-label"
            >
              Email
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="email"
                id="email"
                name="email"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-17 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-19 PrivateNotchedOutline-legendNotched-20"
                >
                  <span>
                    Email
                  </span>
                </legend>
              </fieldset>
            </div>
            
          </div>
        </div>
        <div
          class="form-data__label"
        >
          <select
            class="form-data__label select-component"
            data-testid="gender"
            name="gender"
          >
            <option
              value=""
            >
              - SELECT ONE -
            </option>
            <option
              value="male"
            >
              Male
            </option>
            <option
              value="female"
            >
              Female
            </option>
            <option
              value="other"
            >
              Other
            </option>
          </select>
        </div>
        <div
          class="form-data__submit submit"
        >
          <input
            class="submit__button"
            data-testid="submit"
            disabled=""
            id="submit"
            type="submit"
            value="Submit"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Form validation should firstName be required for submitting 1`] = `
<div>
  <div
    class="my-form"
  >
    <pre>
      <code />
    </pre>
    <div
      class="my-form__message"
    />
    <div
      class="my-form__title"
    >
      üìù Register Form
    </div>
    <div
      class="my-form__box"
    >
      <form
        class="form-data"
      >
        <div
          class="form-data__label"
        >
          <div
            class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-6"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-error Mui-error Mui-focused Mui-focused"
              data-shrink="true"
              for="firstName"
              id="firstName-label"
            >
              First name
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error Mui-focused Mui-focused MuiInputBase-formControl"
            >
              <input
                aria-describedby="firstName-helper-text"
                aria-invalid="true"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="firstName"
                id="firstName"
                name="firstName"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-9 PrivateNotchedOutline-legendNotched-10"
                >
                  <span>
                    First name
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error Mui-focused"
              id="firstName-helper-text"
            >
              <span
                data-testid="firstname-error-message"
              >
                First name is required
              </span>
            </p>
          </div>
        </div>
        <div
          class="form-data__label"
        >
          <div
            class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-6"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink="true"
              for="lastName"
              id="lastName-label"
            >
              Last name
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="lastName"
                id="lastName"
                name="lastName"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-9 PrivateNotchedOutline-legendNotched-10"
                >
                  <span>
                    Last name
                  </span>
                </legend>
              </fieldset>
            </div>
            
          </div>
        </div>
        <div
          class="form-data__label"
        >
          <div
            class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-6"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-error Mui-error"
              data-shrink="false"
              for="phone"
              id="phone-label"
            >
              Phone
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-formControl"
            >
              <input
                aria-describedby="phone-helper-text"
                aria-invalid="true"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="phone"
                id="phone"
                name="phone"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-9"
                >
                  <span>
                    Phone
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error"
              id="phone-helper-text"
            >
              Phone is required
            </p>
          </div>
        </div>
        <div
          class="form-data__label"
        >
          <div
            class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-6"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-error Mui-error"
              data-shrink="false"
              for="email"
              id="email-label"
            >
              Email
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-formControl"
            >
              <input
                aria-describedby="email-helper-text"
                aria-invalid="true"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="email"
                id="email"
                name="email"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-9"
                >
                  <span>
                    Email
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error"
              id="email-helper-text"
            >
              Email is required
            </p>
          </div>
        </div>
        <div
          class="form-data__label"
        >
          <select
            class="form-data__label select-component"
            data-testid="gender"
            name="gender"
          >
            <option
              value=""
            >
              - SELECT ONE -
            </option>
            <option
              value="male"
            >
              Male
            </option>
            <option
              value="female"
            >
              Female
            </option>
            <option
              value="other"
            >
              Other
            </option>
          </select>
          <span
            role="alert"
          >
            Gender is required
          </span>
        </div>
        <div
          class="form-data__submit submit"
        >
          <input
            class="submit__button"
            data-testid="submit"
            id="submit"
            type="submit"
            value="Submit"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Form validation should lastName be required for submitting 1`] = `
<div>
  <div
    class="my-form"
  >
    <pre>
      <code />
    </pre>
    <div
      class="my-form__message"
    />
    <div
      class="my-form__title"
    >
      üìù Register Form
    </div>
    <div
      class="my-form__box"
    >
      <form
        class="form-data"
      >
        <div
          class="form-data__label"
        >
          <div
            class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-11"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink="true"
              for="firstName"
              id="firstName-label"
            >
              First name
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="firstName"
                id="firstName"
                name="firstName"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-14 PrivateNotchedOutline-legendNotched-15"
                >
                  <span>
                    First name
                  </span>
                </legend>
              </fieldset>
            </div>
            
          </div>
        </div>
        <div
          class="form-data__label"
        >
          <div
            class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-11"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-error Mui-error Mui-focused Mui-focused"
              data-shrink="true"
              for="lastName"
              id="lastName-label"
            >
              Last name
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error Mui-focused Mui-focused MuiInputBase-formControl"
            >
              <input
                aria-describedby="lastName-helper-text"
                aria-invalid="true"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="lastName"
                id="lastName"
                name="lastName"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-14 PrivateNotchedOutline-legendNotched-15"
                >
                  <span>
                    Last name
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error Mui-focused"
              id="lastName-helper-text"
            >
              <span
                data-testid="lastname-error-message"
              >
                Last name is required
              </span>
            </p>
          </div>
        </div>
        <div
          class="form-data__label"
        >
          <div
            class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-11"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-error Mui-error"
              data-shrink="false"
              for="phone"
              id="phone-label"
            >
              Phone
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-formControl"
            >
              <input
                aria-describedby="phone-helper-text"
                aria-invalid="true"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="phone"
                id="phone"
                name="phone"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-14"
                >
                  <span>
                    Phone
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error"
              id="phone-helper-text"
            >
              Phone is required
            </p>
          </div>
        </div>
        <div
          class="form-data__label"
        >
          <div
            class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-11"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-error Mui-error"
              data-shrink="false"
              for="email"
              id="email-label"
            >
              Email
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-formControl"
            >
              <input
                aria-describedby="email-helper-text"
                aria-invalid="true"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="email"
                id="email"
                name="email"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-14"
                >
                  <span>
                    Email
                  </span>
                </legend>
              </fieldset>
            </div>
            <p
              class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error"
              id="email-helper-text"
            >
              Email is required
            </p>
          </div>
        </div>
        <div
          class="form-data__label"
        >
          <select
            class="form-data__label select-component"
            data-testid="gender"
            name="gender"
          >
            <option
              value=""
            >
              - SELECT ONE -
            </option>
            <option
              value="male"
            >
              Male
            </option>
            <option
              value="female"
            >
              Female
            </option>
            <option
              value="other"
            >
              Other
            </option>
          </select>
          <span
            role="alert"
          >
            Gender is required
          </span>
        </div>
        <div
          class="form-data__submit submit"
        >
          <input
            class="submit__button"
            data-testid="submit"
            id="submit"
            type="submit"
            value="Submit"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
